src/protocols/secure_channel/CASESession.cpp

```diff
@@ -198,6 +204,12 @@ CHIP_ERROR CASESession::FromSerializable(const CASESessionSerializable & seriali
     memcpy(mMessageDigest, serializable.mMessageDigest, length);
 
     SetPeerNodeId(LittleEndian::HostSwap64(serializable.mPeerNodeId));
+    Credentials::CATValues peerCATs;
+    for (size_t i = 0; i < serializable.mPeerCATs.size(); i++)
+    {
+        peerCATs.val[i] = LittleEndian::HostSwap32(serializable.mPeerCATs.val[i]);
+    }
+    SetPeerCATs(peerCATs);
```

Comment:
How do we determine the number? Not every cert will have max number of CATs. We should make sure that we only ever store `kUnderfinedCat` in the unused slots. is that guaranteed?

Reply:
Yes, it is guaranteed by the code. The only function that sets the CAT values is `ExtractCATsFromOpCert()`, which sets CAT values to `kUnderfinedCat` is the value is not present.